### PR TITLE
Add requires for min/max clusters to test framework

### PR DESCRIPTION
**Pull request by nmittler**
This also detangles some dependencies required to make the suite test work.

### CODE REVIEW

File: pkg/test/framework/suite.go [hunk 1]
```diff
@@ -108,16 +107,64 @@ func (s *Suite) RequireEnvironment(name environment.Name) *Suite {
...
+			return fmt.Errorf("failed setup: Required min clusters not available")
+		}
+		return nil
+	}
+
+	s.requireFns = append(s.requireFns, fn)
+	return s
+}
+
+// RequireMinClusters ensures that the current environment contains at least the given number of clusters.
```
Comment (howardjohn) [inline pkg/test/framework/suite.go]: nit: comment is for min not max

File: tests/integration/telemetry/tracing/servertracing/tracing_test.go [hunk 2]
```diff
@@ -58,6 +58,7 @@ func TestProxyTracing(t *testing.T) {
 func TestMain(m *testing.M) {
 	framework.NewSuite("tracing_test", m).
 		RequireEnvironment(environment.Kube).
+		RequireSingleCluster().
```
Comment (howardjohn) [inline tests/integration/telemetry/tracing/servertracing/tracing_test.go]: are all of these expected to be temporary? we probably don't want to have to add this to every single test if not. Maybe `AllowMultiCluster()`? I guess we can see how things turn out

Comment (ayj) [reply tests/integration/telemetry/tracing/servertracing/tracing_test.go]: It looks like the default behavior is one cluster if this isn't added. That seems reasonable. Do we need to re-specify the defaults?

Comment (nmittler) [reply tests/integration/telemetry/tracing/servertracing/tracing_test.go]: Yeah it's expected to be temporary. We'll eventually want to rewrite as many tests as possible to work in any environment. Requiring single cluster was a natural fallout from the min/max requirement.  So I guess you're suggesting that it should require single cluster by default? Seems reasonable.

Comment (howardjohn) [reply tests/integration/telemetry/tracing/servertracing/tracing_test.go]: I think we just want to make sure whatever the "normal" case is doesn't require a bunch of settings. If supporting multi cluster is (intended) to be normal, then I think how it is makes sense. If we expected long term only a couple tests, then it wouldn't make sense to have to put this on every test to me

Comment (nmittler) [reply tests/integration/telemetry/tracing/servertracing/tracing_test.go]: Actually the more I think about it, I think not having a requirement is probably better. We want as many suites/tests to  run on multiple clusters as possible. But just to make things explicit for now, we have all the existing tests opt-out of multicluster testing. WDYT?

Comment (nmittler) [reply tests/integration/telemetry/tracing/servertracing/tracing_test.go]: And yes, I would expect multicluster tests to be the new normal.

Comment (nmittler) [reply tests/integration/telemetry/tracing/servertracing/tracing_test.go]: @ayj 

> It looks like the default behavior is one cluster if this isn't added. That seems reasonable. Do we need to re-specify the defaults?

The default behavior is to not enforce any requirement on the number of clusters.

**Pull request merged**